Home Task №36

mykolamuntian opened the pull request:
Task:
For the FileEntityText project:
- Implement the readAll method for TextFileVisaAccessor.
- * Implement the PassportService interface. 

Solution:
 - To save and to read entities from files create accessors interfaces for each entity and their imlementations.
 - Common code for reading and writing move to abstract accessor which is extended by accessors of entities. In entities accessors implement writing and reading of these entities.
 - Create PassportService interface and its implementation TextFilePassportService to combine work with visas and passports using their interfaces.

Comments to commit:
 - Created TextFilePassportService to save and to read passports with visas from different files.
 - Changed signature of methods save() and readAll in PassportService interface
 - Moved common code from TextFilePassportAccessor and TextFileVisaAccessor to 
    AbstractTextFileAccessor.
 - Implemented method readAll() in TextFileVisaAccessor.

mykolamuntian committed:
- Created TextFilePassportService to save and to read passports
with visas from different files
 - Changed signature of methods save() and readAll in PassportService
interface
 - Moved common code from TextFilePassportAccessor and
 TextFileVisaAccessor to AbstractTextFileAccessor
 - Implemented method readAll() in TextFileVisaAccessor

strazhkoe commented on src/main/java/com/muntian/passportinfile/services/TextFilePassportService.java:
```diff
@@ -0,0 +1,42 @@
+
+    @Override
+    public Collection<Passport> readAll(String passportFileName, String visaFileName) {
+        PassportAccessor passportAccessor = new TextFilePassportAccessor(passportFileName);
```
File accessors should be fields and their instances should be passed using constructor.

mykolamuntian committed:
Changed signatures of readAll() and save() methods in PassportService
Moved accessors in field in TextFileService
Fixed Main according to changes in TextFileService

strazhkoe commented on src/main/java/com/muntian/passportinfile/services/TextFilePassportService.java:
```diff
@@ -11,16 +11,25 @@
+    public TextFilePassportService(String passportFileName, String visaFileName) {
+        this.passportAccessor = new TextFilePassportAccessor(passportFileName);
+
+        this.visaAccessor = new TextFileVisaAccessor(visaFileName);
```
Accessor objects should be passed as arguments. It is not good to create them in constructor.

mykolamuntian committed:
Passed accessors into constructor of TextFilePassportService

mykolamuntian committed:
Deleted redundant parameters passportFileName and visaFileName

mykolamuntian committed:
Fixed Main according to changes in TextFilePassportService
